packages/test/test-end-to-end-tests/src/test/sharedStringEndToEndTests.spec.ts

```diff
@@ -68,4 +70,43 @@ describeFullCompat("SharedString", (getTestObjectProvider) => {
         assert.equal(
             newSharedString.getText(), text, "The new container should receive the inserted text on creation");
     });
+
+    it("marker passes on attachment directly and transitively to any referenced DDS", async () => {
+        // Insert a simple marker.
+        sharedString1.insertMarker(
+            0,
+            ReferenceType.Simple,
+            {
+                [reservedMarkerIdKey]: "markerId",
+            },
+        );
+        const detachedString1 = SharedString.create(dataObject1.runtime, "detachedString1");
+        detachedString1.insertMarker(
+            0,
+            ReferenceType.Simple,
+            {
+                [reservedMarkerIdKey]: "markerId",
+            },
+        );
+        const detachedString2 = SharedString.create(dataObject1.runtime, "detachedString2");
+
+        // When an unattached DDS refers to another unattached DDS, both remain unattached
+        const simpleMarker = detachedString1.getMarkerFromId("markerId") as Marker;
+        const prop = { color: detachedString2.handle };
+        detachedString1.annotateMarker(simpleMarker, prop);
+
+        assert.equal(detachedString1.isAttached(), false, "detachedString1 should not be attached");
+        assert.equal(detachedString2.isAttached(), false, "detachedString2 should not be attached");
+        assert.equal(sharedString1.isAttached(), true, "sharedString1 should be attached");
+
+        // When referring SharedString becomes attached, the referred SharedString becomes attached
+        // and the attachment transitively passes to a second referred SharedString
+        const simpleMarker2 = detachedString1.getMarkerFromId("markerId") as Marker;
```

Comment:
Shouldn't this be sharedString1 instead of detachedString1?

Reply:
This might coincidentally work since the markers on different SharedStrings have the same ids.

Reply:
Ah that's why it worked. Got it. Thanks for catching that!